recipes/instinct-cpp/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_duckdb": [True, False],
+        "with_exprtk": [True, False],
+        "with_pdfium": [True, False],
+        "with_duckx": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_duckdb": True,
+        "with_exprtk": True,
+        "with_pdfium": True,
+        "with_duckx": True,
+        "duckdb/*:with_httpfs": True,
+        "icu/*:with_extras": True,
+        "icu/*:data_packaging": "static"
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "15",
+            "clang": "17",
+            "gcc": "12"
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("hash-library/8.0", transitive_headers=True)
+        self.requires("bshoshany-thread-pool/4.1.0", transitive_headers=True)
+
+        if self.options.with_duckdb:
+            self.requires("duckdb/0.10.2", transitive_headers=True)
+
+        self.requires("uriparser/0.9.7", transitive_headers=True)
+        self.requires("crossguid/0.2.2", transitive_headers=True)
+        self.requires("protobuf/5.27.0", transitive_headers=True)
+        self.requires("reactiveplusplus/2.1.1", transitive_headers=True)
+        self.requires("icu/74.1")
+        self.requires("tsl-ordered-map/1.1.0", transitive_headers=True)
+        self.requires("fmt/10.2.1", transitive_headers=True)
+        self.requires("fmtlog/2.2.1", transitive_headers=True)
+        self.requires("nlohmann_json/3.11.3", transitive_headers=True)
+        self.requires("base64/0.5.2", transitive_headers=True)
+        self.requires("libcurl/8.6.0", transitive_headers=True)
+        self.requires("inja/3.4.0", transitive_headers=True)
+        self.requires("concurrentqueue/1.0.4", transitive_headers=True)
+        self.requires("cpptrace/0.5.4", transitive_headers=True)
+
+        if self.options.with_pdfium:
+            self.requires("pdfium/95.0.4629", transitive_headers=True)
+
+        if self.options.with_duckx:
+            self.requires("duckx/1.2.2", transitive_headers=True)
+
+        if self.options.with_exprtk:
+            self.requires("exprtk/0.0.2", transitive_headers=True)
+
+        self.requires("llama-cpp/b3040", transitive_headers=True)
+        self.requires("cpp-httplib/0.15.3", transitive_headers=True)
+        self.requires("cli11/2.4.1")
```

Comment:
I had to remove the gtest test_require from here:

If we want to give users the ability to run the tests, we have a way to control that with the `skip_tests` conf, but usually CCI packages simply just don't build/run them, so no need to add test_requirements :)